torch/distributed/_composable/contract.py

```diff
@@ -190,5 +190,10 @@ def _get_registry(module: nn.Module) -> Dict[str, RegistryItem]:
     Get an ``OrderedDict`` of composable APIs that have been applied to the
     ``module``, indexed by the API name.
     """
-    default_registry: Dict[str, RegistryItem] = OrderedDict()
-    return module.__dict__.setdefault(REGISTRY_KEY, default_registry)  # type: ignore[call-overload]
+    registry_key = getattr(module, REGISTRY_KEY, None)
+    if registry_key is None:
+        default_registry: Dict[str, RegistryItem] = OrderedDict()
+        setattr(module, REGISTRY_KEY, default_registry)
```

Comment:
Does your new version go through module's setattr method, whereas the previous one didn't?  I am not sure if that's important, but sometimes subtle bugs creep in that way

Reply:
I think they are identical, will test.

Reply:
This will make perf worse, getattr/setattr is worse perf than dict ordinarily, but NN module has a fairly complicated `__setattr__` handler which makes it worse. getattr access on NN module is famously slow (remember the conversation about CSE?)

Reply:
I'm also just not that keen on this change, we should be in the business of modeling `__dict__` properly...